sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java

```diff
@@ -42,15 +42,13 @@
   // Array for each type. Only 1 is populated for any type.
   private byte[] byteData;
   private short[] shortData;
+  // This is not only used to store data for int column vector, but also can store offsets and
+  // lengths for array column vector.
   private int[] intData;
```

Comment:
One question I have is, the capacity of `ColumnVector` is bound by `MAX_CAPACITY`. Previously we store offset and length individually, so we can have `MAX_CAPACITY` arrays at most. Now we store offset and length together in data/intData which is bound to `MAX_CAPACITY`, doesn't it say we can just have `MAX_CAPACITY / 2` arrays at most?